chat-commands.js

```diff
@@ -2468,18 +2469,25 @@ exports.commands = {
 				Chat.uncacheTree('./tournaments');
 				global.Tournaments = require('./tournaments');
 				Tournaments.tournaments = runningTournaments;
-
-				return this.sendReply("Chat commands have been hot-patched.");
+				this.sendReply("Chat commands have been hot-patched.");
 			} else if (target === 'tournaments') {
+				if (hotpatchLock['tournaments']) return this.errorReply(`Hot-patching tournaments has been disabled by ${hotpatchLock['tournaments'].by} (${hotpatchLock['tournaments'].reason})`);
+
 				let runningTournaments = Tournaments.tournaments;
 				Chat.uncacheTree('./tournaments');
 				global.Tournaments = require('./tournaments');
 				Tournaments.tournaments = runningTournaments;
-				return this.sendReply("Tournaments have been hot-patched.");
+				this.sendReply("Tournaments have been hot-patched.");
 			} else if (target === 'battles') {
+				if (hotpatchLock['battles']) return this.errorReply(`Hot-patching battles has been disabled by ${hotpatchLock['battles'].by} (${hotpatchLock['battles'].reason})`);
+
 				Rooms.SimulatorProcess.respawn();
-				return this.sendReply("Battles have been hotpatched. Any battles started after now will use the new code; however, in-progress battles will continue to use the old code.");
+				this.sendReply("Battles have been hot-patched. Any battles started after now will use the new code; however, in-progress battles will continue to use the old code.");
 			} else if (target === 'formats') {
+				if (hotpatchLock['formats']) return this.errorReply(`Hot-patching formats has been disabled by ${hotpatchLock['formats'].by} (${hotpatchLock['formats'].reason})`);
+				if (hotpatchLock['battles']) return this.errorReply(`Hot-patching battles has been disabled by ${hotpatchLock['battles'].by} (${hotpatchLock['battles'].reason})`);
+				if (hotpatchLock['validator']) return this.errorReply(`Hot-patching the validator has been disabled by ${hotpatchLock['validator'].by} (${hotpatchLock['validator'].reason})`);
```

Comment:
You should check for `hotpatchLock['formats']` in battle and validator hotpatches, though.